src/main/java/com/fasterxml/jackson/dataformat/xml/deser/FromXmlParser.java

```diff
@@ -932,12 +932,11 @@ public String nextTextValue() throws IOException
         switch (token) {
         case XmlTokenStream.XML_END_ELEMENT:
             if (_mayBeLeaf) {
-                // NOTE: this is different from nextToken() -- produce "", NOT null
```

Comment:
Ok, this explicit comment makes me nervous: it specifically spells out that when asking from `nextTextValue()`, empty element should not be coerced into `null`.

But since all existing unit tests, and new ones too, pass, this may make sense.

One related note: the reason `Map`s work ok is that there is no specialized access to "String-valued maps" (but there is for `String[]` and `Collection<String>`!) to use `JsonParser.nextTextValue()`.

One way to alleviate my concerns is to make sure tests are improved -- I noticed additions which are great.
The main addition I'd like to see is addition of checking of `String[]` / `List<String>` valued POJO properties, with null, "". If those retain values as expected I am comfortable with this change.